### DISCUSSION TITLE
player velocity isn't consistent?

### DISCUSSION THREAD

**Discussion by soup587** [Techokami/SonicWorldsNext#117] [Q&A] [answered]
not completely sure how the player controller works but using the player's velocity (as in just Player.velocity) doesn't seem to keep up at higher speeds, after reaching a high enough speed (400 iirc?) it seems to loop back to 0

**Comment by DimensionWarped** [comment]
Can you provide a demonstration? I've never seen this behavior.

**Reply by soup587** [reply]
![image](https://github.com/Techokami/SonicWorldsNext/assets/41347805/e76b8805-fc69-4f77-8b22-acd9a868bfce)
![image](https://github.com/Techokami/SonicWorldsNext/assets/41347805/fadd9c6c-d1ca-4734-be2c-cf71e0cb09c3)

not a fantastic example, sorry, but here i've got a node2d that's a sibling to the player object in a scene with just this code:

```
extends Node
var playerIs

func _ready():
	playerIs = $"../Player"
func _process(delta):
	print(playerIs.velocity.length())
```

at a default running start it prints a speed of 360, but with a fully charged peelout it's.. 240?
if i go above 400 in any other way it'll print 0 and count up from there as if 400 is 0

**Reply by soup587** [reply]
i've noticed it both using length (when previously attempting dynamic music) and just the raw vector2 value (when trying to make badnik gibs that spawn with the player's velocity)

**Reply by soup587** [reply]
oh, it's 480, not 400

**Reply by Renhoex** [reply, marked as answer]
To get the game running like sonic I had to do a work around, if you want to mess with sonics movements you need to use the `movement` variable instead

**Reply by soup587** [reply]
other than a slight value when pushing against something (which could be accounted for) this works great!
